### PR TITLE
General guidance documentation

### PR DESCRIPTION
**Pull request by pioardi**
## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

Documentation enhancement to drive our users to choose the right pool.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -199,13 +199,25 @@ This method will call the terminate method on each worker.
...
 We already have a bench folder where you can find some comparisons.  
-**Thread pools** ( FixedThreadPool and DynamicThreadPool ) are suggested to run CPU intensive tasks, you can still run I/O intensive tasks into thread pools, but performance enhancement is expected to be minimal.  
+
+Before to jump into each poolifier pool type, let highlight that **Node.js comes with a thread pool already**, the libuv thread pool where some particular tasks already run by default.  
+Please take a look at [which tasks run on the libuv thread pool](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/#what-code-runs-on-the-worker-pool).  
+
+Now **if your task runs on libuv thread pool**, you can :
+
+ -Increase the libuv thread pool size setting the [UV_THREADPOOL_SIZE](https://nodejs.org/api/cli.html#cli_uv_threadpool_size_size) and/or  
+ -Use poolifier cluster pool that spawning child processes will also increase the number of libuv threads since that any new child process comes with a separated libuv thread pool.
```
Comment (pioardi) [inline README.md:209]: I need to be 100% sure ot this statement

Comment (pioardi) [reply README.md:209]: Ok I did another check monitoring the number of total threads with top command and using a pool with 50 processes.
So when I start the process I see much more threads ( aroung 300 more ), when I stop it the number decrease

**Comment by pioardi** [comment]
@jerome-benoit  I think this version it too much verbose now....the version on master is much more easy to follow

**Comment by jerome-benoit** [comment]
> @jerome-benoit I think this version it too much verbose now....the version on master is much more easy to follow

I tend to disagree regarding the content explaining clearly the when to use poolifier and how to choose your worker type. 

Maybe you can just rename the section to: 

+ Do I need poolifier? or When to use poolifier? 
+ How to choose your pool worker type? 
   + ...
   + ...

**Comment by pioardi** [comment]
> > @jerome-benoit I think this version it too much verbose now....the version on master is much more easy to follow
> 
> I tend to disagree regarding the content explaining clearly the when to use poolifier and how to choose your worker type.
> 
> Maybe you can just name the sections to:
> 
> * Do I need poolifier? or When to use poolifier? or Why poolifier?
> * How to choose your pool worker type? or Worker Threads vs Cluster Worker
>   
>   * ...
>   * ...

mmmm yes I also feel we need more clarity on that, even if we already have a why poolifier section....we just cannot explain the whole node js internals into our documentation project this is what I would avoid basically.
Let's do in this way I will wait for a your proposal ( take the time that you need, no problem ) :) I feel you can help a lot on this

**Comment by pioardi** [comment]
@jerome-benoit  so in my understanding you see values in those changes I did some other little adjustments, I would like your review and contribution on that ( in particular on section names and organize the content, more than the content itself, but feel free also to update the content if you want ).

File: README.md [hunk 2]
```diff
@@ -34,7 +34,7 @@
 Poolifier is used to perform CPU intensive and I/O intensive tasks on nodejs servers, it implements worker pools (yes, more worker pool implementations, so you can choose which one fit better for you) using [worker-threads](https://nodejs.org/api/worker_threads.html#worker_threads_worker_threads) and cluster pools using [Node.js cluster](https://nodejs.org/api/cluster.html) modules.  
 With poolifier you can improve your **performance** and resolve problems related to the event loop.  
 Moreover you can execute your tasks using an API designed to improve the **developer experience**.  
-Please consult our guidelines to <a href="#choose-your-pool">choose your pool</a>
+Please consult our  <a href="#general-guidance">general guidelines</a>
```
Comment (Shinigami92) [inline README.md]:
```suggestion
Please consult our <a href="#general-guidance">general guidelines</a>
```

**Review by Shinigami92 — APPROVED** [review]

**Review by jerome-benoit — APPROVED** [review]

**Pull request merged**